src/app/model/PieceMovement.ts

```diff
@@ -0,0 +1,5 @@
+type Board = { squares: Square[] };
```

Comment:
Przypuszczam, że board raczej będzie jakimś interfejsem :) Pomyśl co chcesz się dowiedzieć i zrób takie metody. To, że np. będzie lista squares czy cokolwiek innego to jest szczegół. Na co Board powinien Ci odpowiadać - np. czy ktoś stoi na danym polu? itp.

Reply:
(notka na jutro :-) )
Ok, board jako interfejs z metodami np. isPositionEmpty, onPositionPiece, na szybko nie wiem co jeszcze

Reply:
Zdaje się, że tak :)

Reply:
Jak doszedłem do logiki ruchu wieży...to wyjątkowo mi ten Board jako interfejs nie leży... przecież muszę po czymś iterować!? Zrobiłem go jako matrixa Square[][] i wtedy jest cacy - ale pewnie powiesz, że źle :-( ?

 [Tutaj](https://github.com/nowakprojects/CodersCamp2020.Project.TypeScript.Chess/pull/20/commits/7c9e0a3f0e65b8ef6b25fc27d840631deac83291) trochę pozmieniałem, wiem, że mam błąd w `onPositionPiece` i zapewne nie mogę w `type Square` mieć pola piece?: Piece, ale na tę chwilę tak to widzę.... Potem poprawię, bo muszę odejść od kompa.